sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py

```diff
@@ -277,16 +280,20 @@ def get_configuration_setting(
             raise binascii.Error("Connection string secret has incorrect padding")
 
     @distributed_trace
-    def add_configuration_setting(self, configuration_setting, **kwargs):
-        # type: (ConfigurationSetting, dict) -> ConfigurationSetting
+    def add_configuration_setting(
+        self,
+        configuration_setting,  # type: Union[ConfigurationSetting,SecretReferenceConfigurationSetting,FeatureFlagConfigurationSetting] # pylint: disable=line-too-long
```

Comment:
Do we want to list all of them?

My concern is if in the future we want to add more types, this will increase our maintenance cost.

Reply:
I can change to `ConfigurationSetting`, the other two inherit from `ConfigurationSetting`.